pipelines/SpeciesTreeFromBusco/scripts/fetch_genomes_from_db.py

```diff
@@ -0,0 +1,109 @@
+parser.add_argument('-u', metavar='db_URL', type=str, help="Compara master db URL.", required=True)
+parser.add_argument(
+    '-d', metavar='dump_dir', type=str, help="Genome dump directory.", required=True)
+parser.add_argument(
+    '-s', metavar='ssid', type=str, help="Species set id.", required=False, default=None)
+parser.add_argument(
+    '-c', metavar='cid', type=str, help="Collection id.", required=False, default=None)
+parser.add_argument(
+    '-o', metavar='output', type=str, help="Output tsv.", required=True)
+
+
+def _dir_revhash(gid: int) -> str:
+    """Build directory hash from genome db id."""
+    dir_hash = list(reversed(str(gid)))
+    dir_hash.pop()
+    return '/'.join(dir_hash)
+
+
+def _build_dump_path(row: Row) -> str:
+    gcomp = ""
+    if row.genome_component is not None:
+        gcomp = f"comp{row.genome_component}."
+    gpath = path.join(args.d, _dir_revhash(row.genome_db_id), f"{row.name}.{row.assembly}.{gcomp}soft.fa")
+    return gpath
+
+
+if __name__ == '__main__':
+    args = parser.parse_args()
+
+    db_url = args.u
+    ss_id = args.s
+    if ss_id == "":
+        ss_id = None
+    coll_id = args.c
+    if coll_id == "":
+        coll_id = None
+    engine = create_engine(db_url, future=True)
+
+    if ss_id is None and coll_id is None:
+        sys.stderr.write("Either a collection name or a species set id must be specified!\n")
+        sys.exit(1)
+
+    if ss_id is not None and coll_id is not None:
+        sys.stderr.write("Specify either a collection name or a species set id!\n")
+        sys.exit(1)
+
+    ss_query = f"""
+        SELECT genome_db_id, genome_db.name, assembly, genebuild, strain_name, display_name, genome_component FROM species_set JOIN genome_db USING(genome_db_id) WHERE species_set_id = {ss_id};
+    """
+    c_query = f"""
+        SELECT genome_db_id, genome_db.name, assembly, genebuild,strain_name, display_name, genome_component FROM species_set_header JOIN species_set USING(species_set_id) JOIN genome_db USING(genome_db_id) where species_set_header.name='{coll_id}' AND species_set_header.last_release is NULL AND species_set_header.first_release IS NOT NULL;
+    """
+    query = ss_query
+    if coll_id is not None:
+        query = c_query
+
+    missing_fas = []
+    ofh = open(args.o, "w")
+    with engine.connect() as conn:
+        result = conn.execute(text(query))
+        for row in result:
+            gpath = _build_dump_path(row)
+            fa_name = str(row.name) + "_" + str(row.assembly)
+            dump_exists = path.exists(gpath)
+            if not dump_exists:
+                missing_fas.append(gpath)
+            fields = [str(row.genome_db_id), str(row.name), str(row.assembly), str(row.genebuild), str(row.strain_name), str(row.display_name), str(row.genome_component), gpath, str(dump_exists), fa_name]
+            tsv_row = ",".join(fields) + "\n"
+            ofh.write(tsv_row)
```

Comment:
...and this could simplify the CSV output:
```suggestion
            writer.writerow([row.genome_db_id, row.name, row.assembly, row.genebuild, row.strain_name, row.display_name, row.genome_component, gpath, dump_exists, fa_name])
```